scrabble.js

```diff
@@ -1,8 +1,169 @@
-var Scrabble = function() {};
+var Scrabble = function() {
+  this.Tiles = {
+    A: 1, E: 1, I: 1,O: 1, U: 1, L: 1, N: 1, R: 1, S: 1, T: 1,
+    D: 2, G: 2,
+    B: 3, C: 3, M: 3, P: 3,
+    F: 4, H: 4, V: 4, W: 4, Y: 4,
+    K: 5,
+    J: 8, X: 8,
+    Q: 10, Z: 10
+  };
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+  this.bonus = 50;
+
+  this.score = function(word) {
+    var userInput = word.toUpperCase();
+    var total = 0;
+
+    for (var i = 0; i < userInput.length; i++) {
+      var letterScore = this.Tiles[userInput[i]];
+      total += letterScore;
+    }
+
+    if (userInput.length === 7) {
+      total += this.bonus;
+    }
+
+    return total;
+  };
+
+  this.highestScoreFrom = function(arrayOfWords) {
+    var highestWord = arrayOfWords[0];
+    var highestScore = this.score(highestWord);
+
+    for (var i = 0; i < arrayOfWords.length; i++) {
+      var wordScore = this.score(arrayOfWords[i]);
+      var wordLen = arrayOfWords[i].length;
+      // console.log("current word: " + arrayOfWords[i]);
+      // console.log("current score: " + wordScore);
+      // console.log("current length: " + wordLen);
+
+      if (highestScore < wordScore) {
+        highestWord = arrayOfWords[i];
+        highestScore = wordScore;
+      }
+      else if (highestScore === wordScore && highestWord.length !== 7 && wordLen === 7 ) {
+        highestWord = arrayOfWords[i];
+        highestScore = wordScore;
+      }
+      else if (highestScore === wordScore && highestWord.length !== 7 && highestWord.length > wordLen ) {
+        highestWord = arrayOfWords[i];
```

Comment:
You're using the `arrayOfWords[i]` variable a bunch so it might be good to assign it to a variable to make the code more readable.